scipy/sparse/tests/test_coo.py

```diff
@@ -283,8 +391,463 @@ def test_2d_matmul_multivector():
     res = arr1d @ den_a.T
     assert_equal(res, exp)
 
+    # sparse-dense matmul for self.ndim = 1 and other.ndim = 2
+    den_a = np.array([1, 0, 2])
+    den_b = np.array([[3], [4], [0]])
+    exp = den_a @ den_b
+    res = coo_array(den_a) @ den_b
+    assert_equal(res, exp)
+    res = coo_array(den_a) @ list(den_b)
+    assert_equal(res, exp)
+
 
 def test_1d_diagonal():
     den = np.array([0, -2, -3, 0])
     with pytest.raises(ValueError, match='diagonal requires two dimensions'):
         coo_array(den).diagonal()
+
+
+@pytest.mark.parametrize('shape', [(0,), (1,), (7,), (0,0), (2,0), (4,7),
+                                   (0,0,0), (3,6,2), (5,10,3,13), (1,0,3),])
+def test_nd_todense(shape):
+    np.random.seed(12)
+    arr = np.random.randint(low=0, high=5, size=shape)
+    assert_equal(coo_array(arr).todense(), arr)
+
+
+@pytest.mark.parametrize('shape', [(0,), (1,), (2,), (4,), (7,), (12,),
```

Comment:
Yes, two should definitely be sufficient, especially since we have tested the 1D/2D cases already. The examples are arbitrary, I chose random shapes for them; but I think some consistency may be good, so I've made a few changes - 
5a359e4.